agreement/common_test.go

```diff
@@ -348,15 +348,18 @@ func (l *testLedger) EnsureDigest(c Certificate, quit chan struct{}, verifier *A
 		return
 	}
 
-	select {
-	case <-quit:
-		return
-	case <-l.Wait(r):
-		if !consistencyCheck() {
-			err := fmt.Errorf("Wait channel fired without matching block in round %v", r)
-			panic(err)
+	go func() {
```

Comment:
I think that this function is doing beyond and above what it needs to do.
This is part of the agreement service, not the ledger service. I don't think that we need to verify the ledger is working correctly. In the context of your change, this function could become a no-op.